x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/rules/50_query_rule_put.yml

```diff
@@ -0,0 +1,261 @@
+          type: pinned
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ logstash ]
+          actions:
+            ids:
+              - 'id1'
+          priority: 2
+        - rule_id: query-rule-id1
+          type: pinned
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ elastic ]
+          actions:
+            ids:
+              - 'id1'
+              - 'id2'
+
+  # Finally, add another single rule with no priority. This should be appended to the ruleset.
+  - do:
+      query_rule.put:
+        ruleset_id: test-ruleset
+        rule_id: query-rule-id4
+        body:
+          type: 'pinned'
+          criteria:
+            type: 'exact'
+            metadata: 'query_string'
+            values: [ 'search' ]
+          actions:
+            ids:
+              - 'id2'
+
+  - match: { result: 'updated' }
+
+  - do:
+      query_ruleset.get:
+        ruleset_id: test-ruleset
+
+  - match: { ruleset_id: test-ruleset }
+  - match:
+      rules:
+        - rule_id: query-rule-id2
+          type: pinned
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ kibana ]
+          actions:
+            ids:
+              - 'id3'
+              - 'id4'
+          priority: 1
+        - rule_id: query-rule-id3
+          type: pinned
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ logstash ]
+          actions:
+            ids:
+              - 'id1'
+          priority: 2
+        - rule_id: query-rule-id1
+          type: pinned
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ elastic ]
+          actions:
+            ids:
+              - 'id1'
+              - 'id2'
+        - rule_id: query-rule-id4
+          type: pinned
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ search ]
+          actions:
+            ids:
+              - 'id2'
+
+
+---
+'Create Query Rule - Resource already exists':
+  - do:
+      query_rule.put:
+        ruleset_id: test-query-rule-recreating
+        rule_id: abc
+        body:
+          type: 'pinned'
+          criteria:
+            type: 'exact'
+            metadata: 'query_string'
+            values: [ 'elastic' ]
+          actions:
+            ids:
+              - 'id1'
+          priority: 5
+
+  - match: { result: 'created' }
+
+  - do:
+      query_rule.put:
+        ruleset_id: test-query-rule-recreating
+        rule_id: abc
+        body:
+          type: 'pinned'
+          criteria:
+            type: 'exact'
+            metadata: 'query_string'
+            values: [ 'elastic' ]
+          actions:
+            ids:
+              - 'id2'
+          priority: 3
+
+  - match: { result: 'updated' }
```

Comment:
The `updated` result here is because we're updating an existing query ruleset. I am _guessing_ we want to change this for a new query rule? The alternate would be instead of `created/updated` to return the query rule that was updated (@afoucret had suggested that during the RFC period). WDYT?

Reply:
I think it makes sense to return `created` here because the focus of the API is the query rule.

The alternative of returning the query rule that was updated is sub-optimal IMO for two reasons:

- It diverges from the response format for other query rule(set) modification APIs
- It's harder to parse

Reply:
Yeah, the argument for returning the query rule itself is if we had a POST command that generated the ID. That may come later.

Reply:
Updated: [36f2793](https://github.com/elastic/elasticsearch/pull/109042/commits/36f27939cb554ae0600712925a6399d01d993621)